private/buf/bufsync/syncer.go

```diff
@@ -53,194 +62,287 @@ func newSyncer(
 	options ...SyncerOption,
 ) (Syncer, error) {
 	s := &syncer{
-		logger:             logger,
-		repo:               repo,
-		storageGitProvider: storageGitProvider,
+		logger:                                logger,
+		repo:                                  repo,
+		storageGitProvider:                    storageGitProvider,
+		errorHandler:                          errorHandler,
+		modulesDirsForSync:                    make(map[string]struct{}),
+		commitsToTags:                         make(map[string][]string),
+		branchesToModulesForSync:              make(map[string]map[string]bufmoduleref.ModuleIdentity),
+		modulesToBranchesLastSyncPoints:       make(map[string]map[string]string),
+		modulesIdentitiesToCommitsSyncedCache: make(map[string]map[string]struct{}),
 	}
 	for _, opt := range options {
 		if err := opt(s); err != nil {
 			return nil, err
 		}
 	}
-	return s, nil
-}
-
-// resolveSyncPoints resolves sync points for all known modules for the specified branch,
-// returning all modules for which sync points were found, along with their sync points.
-//
-// If a SyncPointResolver is not configured, this returns an empty map immediately.
-func (s *syncer) resolveSyncPoints(ctx context.Context, branch string) (map[Module]git.Hash, error) {
-	syncPoints := map[Module]git.Hash{}
-	// If resumption is not enabled, we can bail early.
-	if s.syncPointResolver == nil {
-		return syncPoints, nil
-	}
-	for _, module := range s.modulesToSync {
-		syncPoint, err := s.resolveSyncPoint(ctx, module, branch)
-		if err != nil {
-			return nil, fmt.Errorf("resolve sync point for module %q in branch %q: %w", module.String(), branch, err)
-		}
-		if syncPoint != nil {
-			s.logger.Debug(
-				"resolved sync point, will sync after this commit",
-				zap.String("branch", branch),
-				zap.Stringer("module", module),
-				zap.Stringer("syncPoint", syncPoint),
-			)
-			syncPoints[module] = syncPoint
-		} else {
-			s.logger.Debug(
-				"no sync point, syncing all branch",
-				zap.String("branch", branch),
-				zap.Stringer("module", module),
-			)
-		}
-	}
-	return syncPoints, nil
-}
-
-// resolveSyncPoint resolves a sync point for a particular module and branch. It assumes
-// that a SyncPointResolver is configured.
-func (s *syncer) resolveSyncPoint(ctx context.Context, module Module, branch string) (git.Hash, error) {
-	syncPoint, err := s.syncPointResolver(ctx, module.RemoteIdentity(), branch)
-	if err != nil {
-		return nil, fmt.Errorf("resolve syncPoint for module %s: %w", module.RemoteIdentity().IdentityString(), err)
-	}
-	if syncPoint == nil {
-		return nil, nil
+	if s.moduleDefaultBranchGetter == nil {
+		s.logger.Warn(
```

Comment:
I refrained myself of doing `logger.Sugar().Warn(` since the output seemed to be more confusing:

With sugar:
```
bufl alpha repo sync --timeout 0 --all-branches \
  --module module-a \
  --module module-b \
  --create --create-visibility public

...

WARN	read module from HEAD failed, module won't be synced for this branch{error 26 0  read module in branch newcontent, commit 890b1d857a468e788feea6c34c26b7e2168ed75d, directory module-b: module not found}

WARN	default branch validation skipped{default_git_branch 15 0 main <nil>} {module 15 0 bufbuild.internal/jfigueroa/module-a <nil>} {error 26 0  BSR module does not exist}

...
```

Without sugar:
```
bufl alpha repo sync --timeout 0 --all-branches \
  --module module-a \
  --create --create-visibility public

...

WARN	read module from HEAD failed, module won't be synced for this branch	{"error": "read module in branch newcontent, commit 890b1d857a468e788feea6c34c26b7e2168ed75d, directory module-a: module not found"}

WARN	default branch validation skipped	{"default_git_branch": "main", "module": "bufbuild.internal/jfigueroa/module-a", "error": "BSR module does not exist"}

...
```